projects/schematics/src/shared/utils/feature-utils.ts

```diff
@@ -484,3 +484,29 @@ function createDependentFeaturesLog(
     ', '
   )}: ${notSelectedFeatures.join(', ')}\n`;
 }
+
+/**
+ * This function is a workaround for the GitHub issue
+ * https://github.com/angular/angular-cli/issues/16320.
+ * Using `ng add` with options of type array (--features)
+ * may not work correctly in some situations, and might only work on the
+ * second attempt.
+ */
+export function getFeaturesOptions<OPTIONS extends LibraryOptions>(
+  options: OPTIONS
+): string[] {
+  return getArrayOptions(options.features ?? []);
+}
```

Comment:
nitpick: Did you consider having a different signature of this util function?
Now we use it as:
`options.features = getFeaturesOptions(options);`, where `getFeaturesOptions` is a pure function that knows about the shape of `OPTIONS`

IMHO more semantic would be using it as:
a) `options.features = convertToArray(options.features)`, where `convertToArray` is a pure function, which doesn't know about the shape of the `OPTIONS` object
OR
b) `normalizeOptionsFeatures(options)` which under the hood mutates the `options.features` property
b) no

Reply:
Thanks, I went with option B as I think it will work better with JSDoc.